src/SessionSetup/ConnectToStadiaWidget.cpp

```diff
@@ -118,17 +123,28 @@ void ConnectToStadiaWidget::SetConnection(StadiaConnection connection) {
       });
 }
 
-void ConnectToStadiaWidget::Start() {
+ErrorMessageOr<void> ConnectToStadiaWidget::Start() {
   if (ssh_connection_artifacts_ == nullptr) {
-    ERROR("Unable to start ConnectToStadiaWidget: ssh_connection_artifacts_ is nullptr");
-    return;
+    std::string error{
+        "Internal error: Unable to start ConnectToStadiaWidget, ssh_connection_artifacts_ is not "
+        "set."};
+    ui_->radioButton->setToolTip(QString::fromStdString(error));
+    setEnabled(false);
+    return ErrorMessage(error);
   }
 
-  auto client_result = orbit_ggp::Client::Create(this);
+  ErrorMessageOr<QPointer<orbit_ggp::Client>> client_result =
+      ErrorMessage("Internal error: Unable to create orbit_ggp::Client instance");
+  if (ggp_executable_path_.isEmpty()) {
+    client_result = orbit_ggp::Client::Create(this);
+  } else {
+    client_result = orbit_ggp::Client::Create(this, ggp_executable_path_);
+  }
```

Comment:
I'm not sure if it is worth changing - especially since it has been like that before, but it's probably worth mentioning:
For proper dependency injection we shouldn't create the `orbit_ggp::Client` in `Start()`. `Start()` should rather accept a `std::unique_ptr<orbit_ggp::Client>`. (`Start()` can then set a parent, and call `release` on the unique_ptr to transfer ownership to Qt's parent-child-model.)